challenge_3/go/makernaren/max_occurance.go

```diff
@@ -0,0 +1,38 @@
+// Take an array of mixed data types and find the element that occurs
+// most number of times.
+package main
+
+import (
+	"fmt"
+)
+
+// Finds the maximum occurance, takes in interface as input and output params
+//as elements in the given array may be of any data type
+func FindMajority(input []interface{}) (result interface{}) {
+	// Create a hashmap, where element of that array is key and its number of
+	// occurances is the value.
+	occurances := make(map[interface{}]int)
+	for _, key := range input {
+		occurances[key]++
+	}
+	// Return when we find the max occurance.
+	max := 0
```

Comment:
You can calculate this on the fly in the loop where you calculate occurrences. Just track what the current item / max count is and update that if the element you just incremented is larger.